### PR TITLE
Introduce Update class

### PR DESCRIPTION
**Pull request by elharo**
@garrettjonesgoogle This refactoring is a step on the way to more configurable reports including upper bounds.

### CODE REVIEW

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+package com.google.cloud.tools.opensource.dependencies;
+
+import java.util.Objects;
+
+import org.eclipse.aether.artifact.Artifact;
+
+/**
+ * Holder for a suggested dependency update. 
+ */
+public class Update {
```
Comment (garrettjonesgoogle) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java]: Have you considered using [AutoValue](https://github.com/google/auto/tree/master/value)? It reduces a lot of boilerplate.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/ArtifactsTest.java [hunk 2]
```diff
@@ -34,5 +34,5 @@ public void testToCoordinates_withPackaging() {
     Artifact artifact = new DefaultArtifact("com.google.example", "example", "jar", "1.3.2");
     Assert.assertEquals("com.google.example:example:1.3.2", Artifacts.toCoordinates(artifact));
   }
-
+
```
Comment (garrettjonesgoogle) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/ArtifactsTest.java]: Undo whitespace regression?

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java (hunk 1 above)
Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java]: Yes, and rejected it for multiple reasons including excess dependencies and idiosyncratic code.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/ArtifactsTest.java (hunk 2 above)
Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/ArtifactsTest.java]: https://groups.google.com/a/google.com/forum/#!msg/java-users/_ABiES5o020/J3TVUclzN2gJ

Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/ArtifactsTest.java]: Official Google Java code review policy is to never comment on trailing white space. Critique doesn't even show it. Unfortunately Github does, but we should still not worry about it.

File: dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java (hunk 1 above)
Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/Update.java]: Also we need a specific toString method.

**Pull request merged**